README.md

```diff
@@ -1,2 +1,44 @@
 # ai_graph_coloring
+
 Artificial intelligence project to color graphs
+
+## Local development
+
+### Requirements
+
+python 2.7
+unix style operating system
+
+### Environment Setup
+
+#### Installing virtualenv
+
+    $ pip install virtualenv
+
+#### Creating a virtual environment
+
+    $ virtualenv <path>
+    
+#### Activating your virtual environment
+
+    $ source <path>/bin/activate
+    
+#### Installing requirements
+
+1. Activate your virtual environment
+
+2. `$ pip install requirements-dev.txt`
+
+#### Adding a git hook
+
+From the root directory of your project:
```

Comment:
More explanation on what this is doing would be good. e.g. "This will make git run tests before you make a commit"